pkg/controller/node/rate_limited_queue.go

```diff
@@ -221,7 +219,7 @@ func (q *RateLimitedTimedQueue) Try(fn ActionFunc) {
 			val.ProcessAt = now.Add(wait + 1)
 			q.queue.Replace(val)
 		} else {
-			q.queue.RemoveFromQueue(val.Value)
+			q.queue.Remove(val.Value)
```

Comment:
RemoveFromQueue() will only removes val.value (which is actually a string in format "{pod.Namespace}/{pod.Name}" in the eviction case) from q.queue, but leave it in q.set, so a new pod with the same val.value can't be added to queue unless Remove(val.value) is called.
See also the change I made on the description of Try().
>
```diff
diff --git a/pkg/controller/node/rate_limited_queue.go b/pkg/controller/node/rate_limited_queue.go
index 7cb258d..f4a3b75 100644
--- a/pkg/controller/node/rate_limited_queue.go
+++ b/pkg/controller/node/rate_limited_queue.go
@@ -195,9 +195,7 @@ type ActionFunc func(TimedValue) (bool, time.Duration)
 // Try processes the queue. Ends prematurely if RateLimiter forbids an action and leak is true.
 // Otherwise, requeues the item to be processed. Each value is processed once if fn returns true,
 // otherwise it is added back to the queue. The returned remaining is used to identify the minimum
-// time to execute the next item in the queue. The same value is processed only once unless
-// Remove is explicitly called on it (it's done by the cancelPodEviction function in NodeController
-// when Node becomes Ready again)
+// time to execute the next item in the queue.
 // TODO: figure out a good way to do garbage collection for all Nodes that were removed from
 // the cluster.
 func (q *RateLimitedTimedQueue) Try(fn ActionFunc) {
```

An example case of to show before & after of this change:
1. say a pod P is running on node N1
2. node N1 goes unreachable, nodecontroller add taint `unreachable:NoExecute` to node N1
3. nodecontroller delete pod P due to NoExecute taint tolerations not match
4. someone created another pod **in the same namespace with the same podName** to prior deleted pod P, let me call it P'
5. pod P' is scheduled onto node N2
6. node N2 goes unreachable, nodecontroller add taint `unreachable:NoExecute` to node N2

Without this change, pod P' can't be queued for eviction due to the dirty data of pod P left in q.set.
With this change, pod P' can be queued and evicted as expected.